python/foglamp/plugins/south/common/sensortag_cc2650.py

```diff
@@ -136,24 +136,31 @@ class SensorTagCC2650(object):
     is_connected = False
 
     def __init__(self, bluetooth_adr, timeout):
-        try:
-            self.bluetooth_adr = bluetooth_adr
-
-            self.con = pexpect.spawn('gatttool -b ' + bluetooth_adr + ' --interactive')
-            self.con.expect('\[LE\]>', timeout=int(timeout))
-
-            msg_debug = 'SensorTagCC2650 {} Connecting... If nothing happens, please press the power button.'.\
-                        format(self.bluetooth_adr)
-            print(msg_debug)
-            _LOGGER.debug(msg_debug)
+        attempt_count = 1
+        while attempt_count <= 5:
+            try:
+                self.bluetooth_adr = bluetooth_adr
+
+                self.con = pexpect.spawn('gatttool -b ' + bluetooth_adr + ' --interactive')
+                self.con.expect('\[LE\]>', timeout=int(timeout))
+
+                msg_debug = 'SensorTagCC2650 {} Connecting... If nothing happens, please press the power button. Attempt no. {}'.\
+                            format(self.bluetooth_adr, attempt_count)
+                print(msg_debug)
+                _LOGGER.debug(msg_debug)
+
+                self.con.sendline('connect')
+                self.con.expect('.*Connection successful.*\[LE\]>', timeout=int(timeout))
+                self.is_connected = True
+                msg_success = 'SensorTagCC2650 {} connected successfully'.format(self.bluetooth_adr)
+                print(msg_success)
+                _LOGGER.debug(msg_success)
+                break
+            except (ExceptionPexpect, EOF, TIMEOUT, Exception) as ex:
+                attempt_count += 1
+                time.sleep(1)
```

Comment:
I would be tempted to backoff the sleeps here - the problem of course is that the init method will block and eventually the monitor will kick in and declare the service as dead. Not sure we have a good solution for this in the poll mode. It may be we have to say the poll mode can not work with gatttool like this.

Reply:
Reduced the sleep time from 1 second to 0.5 second. Hope this is acceptable. Secondly, the alternate to gatttool is hcitool. May be we should experiment with hcitool in next sprint.